numpy/testing/_private/pytesttester.py

```diff
@@ -0,0 +1,329 @@
+
+        Returns
+        -------
+        result : bool
+            Return True on success, false otherwise.
+
+        Notes
+        -----
+        Each NumPy module exposes `test` in its namespace to run all tests for it.
+        For example, to run all tests for numpy.lib:
+
+        >>> np.lib.test() #doctest: +SKIP
+
+        Examples
+        --------
+        >>> result = np.lib.test() #doctest: +SKIP
+        Running unit tests for numpy.lib
+        ...
+        Ran 976 tests in 3.933s
+
+        OK
+
+        >>> result.errors #doctest: +SKIP
+        []
+        >>> result.knownfail #doctest: +SKIP
+        []
+
+        """
+        import pytest
+
+        # cap verbosity at 3, which is equivalent to the pytest '-vv' option
+        verbose = min(int(verbose), 3)
+
+        #FIXME This is no longer needed? Assume it was for use in tests.
+        from . import utils
+        utils.verbose = verbose
+
+        # setup the pytest arguments
+        pytest_args = ['-l']
+
+        if doctests:
+            raise ValueError("Doctests not supported")
+
+        if extra_argv:
+            pytest_args += list(extra_argv)
+
+        if verbose and int(verbose) > 1:
+            pytest_args += ["-" + "v"*(int(verbose)-1)]
+
+        if coverage:
+            pytest_args += ["--cov=" + self.package_path]
+
+        if label == "fast":
+            pytest_args += ["-m", "not slow"]
+        elif label != "full":
+            pytest_args += ["-m", label]
+
+        if timer and int(timer) > 0:
+            pytest_args += ["--durations=%s" % int(timer)]
+
+        if tests is None:
+            tests = [self.package_name]
+
+        pytest_args += ['--pyargs'] + list(tests)
+
+
+        # setup the warnings argument
+        if raise_warnings is None:
+            raise_warnings = self.raise_warnings
+
+        _warn_opts = dict(develop=(Warning,),
+                          release=())
+        if isinstance(raise_warnings, basestring):
+            raise_warnings = _warn_opts[raise_warnings]
+
+        # run tests.
+        _show_system_info()
+        print(pytest_args)
+
+        with suppress_warnings("location") as sup:
+            # Reset the warning filters to the default state,
+            # so that running the tests is more repeatable.
+            warnings.resetwarnings()
+            # Set all warnings to 'warn', this is because the default 'once'
+            # has the bad property of possibly shadowing later warnings.
+            warnings.filterwarnings('always')
+            # Force the requested warnings to raise
+            for warningtype in raise_warnings:
+                warnings.filterwarnings('error', category=warningtype)
+            # Filter out annoying import messages.
+            sup.filter(message='Not importing directory')
+            sup.filter(message="numpy.dtype size changed")
+            sup.filter(message="numpy.ufunc size changed")
+            sup.filter(category=np.ModuleDeprecationWarning)
```

Comment:
These were mostly from Cython. IIRC, there has been some change on the Cython end, but I don't recall what it was.

Reply:
The warnings are still there I believe: https://github.com/cython/cython/blob/2032632293ccda9d3700f10c85c6be101eeb53f6/Cython/Utility/ImportExport.c#L390

Reply:
They haven't shown up in the last SciPy release, but I suspect that is because we haven't changed `dtype` and `ufunc` sizes in a while. Anyone know a handy way to check that?